stwcs/wcsutil/headerlet.py

```diff
@@ -2017,6 +2015,8 @@ def apply_as_primary(self, fobj, attach=True, archive=True, force=False):
                         alt_hlethdu.append(alt_hlet_hdu)
                         hdrlet_extnames.append(hname)
 
+                        altwcs.deleteWCS(fobj, sciext_list, wcskey=wcskey)
```

Comment:
I think this call to 'deleteWCS' is at the wrong indentation level.  It should be called for ALL wcskey values, right?  So, it needs to be backed out one level, since it currently will only delete altWCS solutions which were not already archived as headerlet extensions.

Reply:
yes, good catch.

Reply:
fixed